pkg/agent/externalnode/external_node_controller.go

```diff
@@ -0,0 +1,753 @@
+		klog.ErrorS(err, "Failed to move configuration from the host interface to uplink", "hostInterface", hostIfName, "uplink", uplinkName)
+		return nil, err
+	}
+	klog.V(2).InfoS("Moved configuration from the uplink to host port", "uplink", uplinkName, "hostInterface", hostIfName)
+	if err := c.ofClient.InstallVMUplinkFlows(hostIfName, hostOFPort, uplinkOFPort); err != nil {
+		return nil, err
+	}
+	klog.InfoS("Added uplink and host port on OVS and installed openflow entries", "uplink", uplinkName, "hostInterface", hostIfName)
+	success = true
+	ifIPs := make([]net.IP, 0)
+	for _, ip := range ips {
+		ifIPs = append(ifIPs, net.ParseIP(ip))
+	}
+	hostIFConfig := &interfacestore.InterfaceConfig{
+		Type:          interfacestore.ExternalEntityInterface,
+		InterfaceName: hostIfName,
+		IPs:           ifIPs,
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: hostIfUUID,
+			OFPort:   hostOFPort,
+		},
+		EntityInterfaceConfig: &interfacestore.EntityInterfaceConfig{
+			EntityName:      eeName,
+			EntityNamespace: eeNamespace,
+			UplinkPort: &interfacestore.OVSPortConfig{
+				PortUUID: uplinkUUID,
+				OFPort:   uplinkOFPort,
+			},
+			HostIfaceIndex: hostIF.Index,
+		},
+	}
+	return hostIFConfig, nil
+}
+
+func getOVSAttachInfo(uplinkName, uplinkUUID, hostIFIdx, hostPortName, entityName, entityNamespace string, ips []string) map[string]interface{} {
+	attachInfo := map[string]interface{}{
+		interfacestore.AntreaInterfaceTypeKey: interfacestore.AntreaHost,
+	}
+	if uplinkName != "" {
+		attachInfo[ovsExternalIDUplinkName] = uplinkName
+	}
+	if uplinkUUID != "" {
+		attachInfo[ovsExternalIDUplinkPort] = uplinkUUID
+	}
+	if hostIFIdx != "" {
+		attachInfo[ovsExternalIDHostIFIndex] = hostIFIdx
+	}
+	if hostPortName != "" {
+		attachInfo[ovsExternalIDHostIFName] = hostPortName
+	}
+	if entityName != "" {
+		attachInfo[ovsExternalIDEntityName] = entityName
+	}
+	if entityNamespace != "" {
+		attachInfo[ovsExternalIDEntityNamespace] = entityNamespace
+	}
+	if len(ips) != 0 {
+		attachInfo[ovsExternalIDIP] = strings.Join(ips, eeSplitter)
+	}
+
+	return attachInfo
+}
+
+func (c *ExternalNodeController) updateOVSPortsData(interfaceConfig *interfacestore.InterfaceConfig, eeName string, ips []string) (*interfacestore.InterfaceConfig, error) {
+	portUUID := interfaceConfig.PortUUID
+	portName := interfaceConfig.InterfaceName
+	port, err := c.ovsBridgeClient.GetPortData(portUUID, portName)
+	if err != nil {
+		return nil, err
+	}
+
+	attachInfo := map[string]interface{}{
+		ovsExternalIDUplinkName:               port.ExternalIDs[ovsExternalIDUplinkName],
+		ovsExternalIDHostIFIndex:              port.ExternalIDs[ovsExternalIDHostIFIndex],
+		ovsExternalIDHostIFName:               port.ExternalIDs[ovsExternalIDHostIFName],
+		ovsExternalIDUplinkPort:               port.ExternalIDs[ovsExternalIDUplinkPort],
+		ovsExternalIDEntityName:               eeName,
+		ovsExternalIDEntityNamespace:          port.ExternalIDs[ovsExternalIDEntityName],
```

Comment:
Agreed. Since namespace is tied to the antrea-agent.conf, it wont change during runtime. 
The reason I asked this, windows agent was relying on this function to update ovsExternalIDEntityName and ovsExternalIDEntityNamespace, that needs to be changed now.

Reply:
We can update the namespace in the windows initializer. But not the ExternalEntity name
Do you mean, it should generate the name based of ExternalNode name and interface name?

Reply:
> We can update the namespace in the windows initializer. But not the ExternalEntity name Do you mean, it should generate the name based of ExternalNode name and interface name?

Yes, I think we have enough information to do it in initializer directly.